src/components/BlockCountry/index.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react'
+import { useModal } from '@pancakeswap/uikit'
+import Cookies from 'js-cookie'
+import { BLOCK_COUNTRIES_COOKIE_NAME } from 'config/constants/cookie-names'
+import BlockCountryModal from './BlockCountryModal'
+
+function BlockCountry() {
+  const [showModal] = useModal(<BlockCountryModal />, false)
+  const showModalRef = React.useRef(showModal)
```

Comment:
Why use ref here and not just showModal? 🤔

Reply:
Oh because the `showModal` from `useModal` is not memoized, it would cause the `useEffect` below unnecessarily re-executed.

By `useRef`, the `useEffect` below would only be executed once.